gulpfile.js

```diff
@@ -72,56 +80,74 @@ const browserifyConfigs = [
 const jsGlob = `./\{${dirs.join(',')}\}/**/*.js`;
 const jsGlobIgnore = `./\{${symlinkDirs.concat(copyDirs).join(',')}\}/**/*.js`;
 
-function onError(err) {
-	gutil.log(gutil.colors.red('Error:'), err);
-	this.emit('end');
+function onError(shouldExit, err) {
+	if(shouldExit) {
+		isExiting = true;
+		killAllWorkers();
+		throw new Error(err);
+	} else {
+		gutil.log(gutil.colors.red('Error:'), err);
+		this.emit('end');
+	}
 }
 
 function onSuccess(msg) {
-	gutil.log(gutil.colors.green('Build:'), msg);
+	if(!isExiting) {
+		gutil.log(gutil.colors.green('Build:'), msg);
+	}
 }
 
-function getBrowserify() {
+function getBrowserify(exitOnError = true) {
 	const streams = browserifyConfigs.map(config => {
 		return gulp
 			.src(config.src)
 			.pipe(tap(file => {
 				file.contents = browserify(file.path, config.config).bundle();
 			}))
 			.pipe(rename(config.dest))
+			.on('error', partial(onError, exitOnError))
```

Comment:
No, this wouldn't work, it would swallow the actual error message.

Partial is used here because `onError` needs to be executed in the context of current stream (hence `apply()` wouldn't work, it needs to know whether to exit and finally it needs to be called with the actual error message.

Replacement code, without the use of `partial` would be much less readable, something like this:

```
.on('error', function(err) { onError.bind(this)(exitOnError, err); })
```